internal/service/advancedclustertpf/validate_schema.go

```diff
@@ -0,0 +1,77 @@
+package advancedclustertpf
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var defaultMongoDBMajorVersion = "8.0"
+
+func PlanMustUseMongoDBVersion(version float64, mustBeLower bool) FailOnIncompatibleMongoDBVersion {
+	return FailOnIncompatibleMongoDBVersion{
+		Version:     version,
+		MustBeLower: mustBeLower,
+	}
+}
+
+type FailOnIncompatibleMongoDBVersion struct {
+	Version     float64
+	MustBeLower bool
+}
+
+func (v FailOnIncompatibleMongoDBVersion) Description(_ context.Context) string {
+	if v.MustBeLower {
+		return fmt.Sprintf("can only be configured if the mongo_db_major_version is %.1f or lower", v.Version)
+	}
+	return fmt.Sprintf("can only be configured if the mongo_db_major_version is %.1f or higher", v.Version)
+}
+
+func (v FailOnIncompatibleMongoDBVersion) MarkdownDescription(ctx context.Context) string {
+	return v.Description(ctx)
```

Comment:
nit: swap and have the main text in MarkdownDescription, it's the main doc method for us

Reply:
Changed in 290b393e5fff8730bcd18f0f501c2d74d3a2469c